app/javascript/components/domain-form/index.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './domain-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (id, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_domains/${id}?attributes=name,description,enabled,source`).then(({
+    name, description, enabled, source,
+  }) => {
+    miqSparkleOff();
+    f.setInitialValues({ name, description, enabled });
+    // FIXME Retrieve readability from the API directly instead of the source (property is missing from the API)
+    f.setDisabled(!['user', 'user_locked'].includes(source));
+  });
+};
+
+const handleSubmit = (id, values, redirectUrl) => {
+  miqSparkleOn(); // miqRedirectBack turns it off
+  const action = id ? 'edit' : 'create';
+  const flashMessage = sprintf(__('Automate Domain "%s" was %s.'),
+    values.name,
+    id ? 'saved' : 'added');
+
+  API.post(`/api/automate_domains/${id || ''}`, {
+    action, ...values,
+  }, { skipErrors: [400] })
+    .then(() => miqRedirectBack(flashMessage, 'success', redirectUrl))
+    .catch(handleFailure);
+};
+
+const NamespaceForm = ({ id }) => {
+  const [initialValues, setInitialValues] = useState({ name: '', description: '', enabled: true });
+  const [disabled, setDisabled] = useState(false);
+  const [loaded, setLoaded] = useState(false);
```

Comment:
You are doing two state updates in one callback (getInitialValues) which will cause multiple re-renders of component. It is recommended to use `useReducer` hook to optimize rendering cycles.